information/specification.md

```diff
@@ -0,0 +1,706 @@
+entries are then _associated with_ that output, and _removed_ from the
+[available UTxO set](#available-utxo-set).
+
+This phase ends when _every_ output has been associated with a selection of
+UTxO entries.
+
+#### Phase 2: Improvement
+
+In this phase, the algorithm attempts to improve upon each of the UTxO
+selections made in the previous phase, by conservatively expanding the
+selection made for each output in order to generate improved change
+values.
+
+During this phase, the algorithm:
+
+  * processes outputs in _ascending order of coin value_.
+
+  * continues to select values from the [available UTxO
+    set](#available-utxo-set).
+
+  * incrementally populates the
+    [accumulated coin selection](#accumulated-coin-selection-1).
+
+For each output of value **_v_**, the algorithm:
+
+ 1.  **Calculates a _target range_** for the total value of inputs used to
+     pay for that output, defined by the triplet:
+
+     (_minimum_, _ideal_, _maximum_) = (_v_, _2v_, _3v_)
+
+ 2.  **Attempts to improve upon the existing UTxO selection** for that output,
+     by repeatedly selecting additional entries at random from the [available
+     UTxO set](#available-utxo-set), stopping when the selection can be
+     improved upon no further.
+
+     A selection with value _v1_ is considered to be an _improvement_ over a
+     selection with value _v0_ if **all** of the following conditions are
+     satisfied:
+
+      * **Condition 1**: we have moved closer to the _ideal_ value:
+
+        abs (_ideal_ − _v1_) < abs (_ideal_ − _v0_)
+
+      * **Condition 2**: we have not exceeded the _maximum_ value:
+
+        _v1_ ≤ _maximum_
+
+      * **Condition 3**: when counting cumulatively across all outputs
+        considered so far, we have not selected more than the _maximum_ number
+        of UTxO entries specified by [Maximum Input
+        Count](#maximum-input-count).
+
+ 3.  **Creates a _change value_** for the output, equal to the total value
+     of the _improved UTxO selection_ for that output minus the value _v_ of
+     that output.
+
+ 4.  **Updates the [accumulated coin
+     selection](#accumulated-coin-selection-1)**:
+
+      * Adds the _output_ to the _outputs_ field;
+      * Adds the _improved UTxO selection_ to the _inputs_ field;
+      * Adds the _change value_ to the _change values_ field.
+
+This phase ends when every output has been processed, **or** when the
+[available UTxO set](#available-utxo-set) has been exhausted, whichever occurs
+sooner.
+
+### Termination
+
+When both phases are complete, the algorithm terminates.
+
+The [accumulated coin selection](#accumulated-coin-selection-1) is returned
+to the caller as the [coin selection](#coin-selection) result.
+
+The [available UTxO set](#available-utxo-set) is returned to the caller as the
+[remaining UTxO set](#remaining-utxo-set) result.
+
+# External Resources
+
+## Self Organization in Coin Selection
+
```

Comment:
Suggestion: add a short summary of the content of this resource, to explain to readers why they should want to read it.

Reply:
@jonathanknowles wrote:
> Suggestion: add a short summary of the content of this resource, to explain to readers why they should want to read it.

Fixed in b778151bdb1eed088fbf83be66b210f43c9b066c.